### DISCUSSION TITLE
[Cloudflare] - 500 Internal Server Error Nginx error for my second website

### DISCUSSION THREAD

**Discussion by Kawaz971** [evertramos/nginx-proxy-automation#364] [Q&A]
Hello,

I have an access issue with me second website : ERROR 500 Nginx.  My first website works well. The Two website are hosted on the same system. 

I want to be able to host two or more websites on my infrastructure.

 - Linux release: Debian Version 11.5
 - Server type Private Server
 - Docker version 20.10.21
 - docker-compose version 2.13.0

 - nginx container : how to get container logs ?
 - letsencrypt container (if related to ssl) : how to get container logs ?

Thanks for your help

**Comment by evertramos** [comment]
Hello @Kawaz971,

If your first domain is running correct, using this nginx-proxy-automation, your second just work just fine, unless you have:

- same domain
- dns issues
- missconfigured the web container

This is not and issue probabely, I will convert it to a discussion.

Please provide your logs as of: 

```bash
docker logs <your-container-name>
```

Please post your docker-compose.yml (without the domain if you prefer) for both websites, and confirm if the domains are differente from both sites.

**Comment by Kawaz971** [comment]
Thanks for your answer here are the logs ==>

docker logs letsencrypt container : 

Creating/renewal sowbella.com certificates... (domaine1.com www.domaine1.com domaine1.fr www.domaine1.fr)
[Mon Dec  5 08:32:35 UTC 2022] Domains not changed.
[Mon Dec  5 08:32:35 UTC 2022] Skip, Next renewal time is: Fri Jan 27 16:22:50 UTC 2023
[Mon Dec  5 08:32:35 UTC 2022] Add '--force' to force to renew.
Creating/renewal teamvava.fr certificates... (domain2.fr www.domain2.fr)
[Mon Dec  5 08:32:36 UTC 2022] Using CA: https://acme-v02.api.letsencrypt.org/directory
[Mon Dec  5 08:32:36 UTC 2022] Creating domain key
[Mon Dec  5 08:32:36 UTC 2022] The domain key is here: /etc/acme.sh/webmaster@domain1.com/domain2.fr/domain2.fr.key
[Mon Dec  5 08:32:36 UTC 2022] Multi domain='DNS:domain2.fr,DNS:www.domain2.fr'
[Mon Dec  5 08:32:36 UTC 2022] Getting domain auth token for each domain
[Mon Dec  5 08:32:38 UTC 2022] Getting webroot for domain='domain2.fr'
[Mon Dec  5 08:32:38 UTC 2022] Getting webroot for domain='www.domain2.fr'
[Mon Dec  5 08:32:38 UTC 2022] Verifying: domain2.fr
[Mon Dec  5 08:32:41 UTC 2022] Pending
[Mon Dec  5 08:32:44 UTC 2022] Pending
[Mon Dec  5 08:32:46 UTC 2022] Pending
[Mon Dec  5 08:32:49 UTC 2022] Pending
[Mon Dec  5 08:32:51 UTC 2022] Pending
[Mon Dec  5 08:32:54 UTC 2022] Pending
[Mon Dec  5 08:32:56 UTC 2022] Pending
[Mon Dec  5 08:32:59 UTC 2022] Pending
[Mon Dec  5 08:33:01 UTC 2022] Pending
[Mon Dec  5 08:33:03 UTC 2022] Pending
[Mon Dec  5 08:33:06 UTC 2022] Pending
[Mon Dec  5 08:33:08 UTC 2022] Pending
[Mon Dec  5 08:33:11 UTC 2022] domain2.fr:Verify error:2606:4700:3031::ac43:992e: Invalid response from http://domain2.fr/.well-known/acme-challenge/xYgQLh79PHAHdd2y7CLnlm6ncLn6m5xYKThANyBJEsA: 522
[Mon Dec  5 08:33:11 UTC 2022] Please check log file for more details: /dev/null
Sleep for 3600s

docker logs docker-gen-auto : 

2022/11/30 00:12:33 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
2022/11/30 00:12:33 Watching docker events
2022/11/30 00:12:33 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
2022/11/30 00:12:33 Received signal: hangup
2022/11/30 00:12:33 Received signal: hangup
2022/11/30 00:12:33 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
2022/11/30 00:12:34 Received signal: hangup
2022/11/30 00:12:34 Received signal: hangup
2022/11/30 00:12:34 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
2022/11/30 00:21:13 Received event start for container fef67f68b31b
2022/11/30 00:21:18 Debounce minTimer fired
2022/11/30 00:21:18 Generated '/etc/nginx/conf.d/default.conf' from 8 containers
2022/11/30 00:21:18 Sending container 'proxy-web-auto' signal '1'
2022/11/30 00:22:10 Received event die for container fef67f68b31b
2022/11/30 00:22:11 Received event stop for container fef67f68b31b
2022/11/30 00:22:16 Debounce minTimer fired
2022/11/30 00:22:16 Generated '/etc/nginx/conf.d/default.conf' from 7 containers
2022/11/30 00:22:16 Sending container 'proxy-web-auto' signal '1'

docker-compose.yml Domain 2:

version: '3'

services:
  new-tv-db:
    container_name: ${CONTAINER_DB_NAME}
    image: ${DB_IMAGE:-mariadb}:${DB_VERSION:-latest}
    restart: unless-stopped
    volumes:
      - ${DB_FILES}:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}

  new-tv-wordpress:
    depends_on:
      - new-tv-db
    container_name: ${CONTAINER_SITE_NAME}
    image: ${SITE_IMAGE:-wordpress}:${SITE_VERSION:-latest}
    restart: unless-stopped
    volumes:
      - ${SITE_FILES}:/var/www/html
      - ./conf.d/php.ini:/usr/local/etc/php/conf.d/php.ini
    environment:
      WORDPRESS_DB_HOST: ${CONTAINER_DB_NAME}:3306
      WORDPRESS_DB_NAME: ${MYSQL_DATABASE}
      WORDPRESS_DB_USER: ${MYSQL_USER}
      WORDPRESS_DB_PASSWORD: ${MYSQL_PASSWORD}
      WORDPRESS_TABLE_PREFIX: ${WORDPRESS_TABLE_PREFIX}
      VIRTUAL_HOST: ${DOMAINS}
      LETSENCRYPT_HOST: ${DOMAINS}
      LETSENCRYPT_EMAIL: ${LETSENCRYPT_EMAIL}
    logging:
      options:
        max-size: ${LOGGING_OPTIONS_MAX_SIZE:-200k}

  wpcli:
    image: tatemz/wp-cli
    volumes:
      - ${SITE_FILES}:/var/www/html
    depends_on:
      - new-tv-db
    entrypoint: wp

networks:
   default:
     external:
       name: ${NETWORK}

**Comment by evertramos** [comment]
It seems your dns is not configured it correctly, if you ping from your server this domain.fr it show your server public ip address?

**Reply by Kawaz971** [reply]
when i ping domaine2.fr i have a well answer. To note, I use the cloudflare service for my domain1.com and for my domain2.fr

**Reply by evertramos** [reply]
That's the point!. When using CloudFlare (I use myself) you have many options, first check this issue #238 to have an idea of the "issue".

1. You may use only cloudflare ssl (tls) and remove the 'LETSENCRYPT' variables from the docker-compose files
2. Create a Server certificate and add it to cloudflare network so you have full tls between cloudflare and your server
3. add a setting in cloudflare allowing to bypass the the knowledge tls connection from letsencrypt to your server
4. remove the proxy in dns (I normally don't recommend that) 

Please let us know if that worked out fine for you.

**Comment by Kawaz971** [comment]
Thank's a lot for your answer. I try applicated your recomandation but i can't do it. 

1. I don't found variables LETSENCRYPT in my file "docker-compose.yml". To try i desabled the docker contenair lestsencrypt-auto
2. I created a server certificate on cloudlfare SSL/TLS ==> Clients Certificates (Host is empty)
3. How to add a setting in cloudflare allowing to bypass the the knowledge tls ?
4. Even by deleting the proxy in dns the problem is the same

as you can understand I always encounter difficulties, I hope that the information I gave you is complete enough

**Reply by evertramos** [reply]
the letsencrypt environment variables would be:

![image](https://user-images.githubusercontent.com/905951/206497635-7e952573-8a9f-49ec-bb2e-6300fde4e85c.png)

and please.. don't stop the proxy services (nginx, letsencrypt and docker-gen) if you stop them all your work behind it will stop working.

**Reply by evertramos** [reply]
@Kawaz971 did you managed to fix it?

**Reply by Kawaz971** [reply]
Unfortunately no ! I have the same error message

**Reply by evertramos** [reply]
I would suggest you come back to the point that you had one working and try again. If one is workin over the proxy, other should work the same way, you are probably missing a details. Please walkthrough the documentation in details. Hope you solve this.
